stan/math/prim/mat/prob/std_binormal_lcdf.hpp

```diff
@@ -0,0 +1,160 @@
+  for (size_t i = 0; i < N_y; ++i) {
+    check_size_match(function,
+                     "Size of one of the vectors "
+                     "of the location variable",
+                     y_vec[i].size(),
+                     "Size of another vector of "
+                     "the location variable",
+                     2);
+  }
+
+  for (size_t n = 0; n < N_y; ++n) {
+    check_not_nan(function, "Random variable", y_vec[n]);
+  }
+
+  for (size_t n = 0; n < N; ++n) {
+    const T_partials_return y1_dbl = value_of(y_vec[n][0]);
+    const T_partials_return y2_dbl = value_of(y_vec[n][1]);
+    const T_partials_return rho_dbl = value_of(rho_vec[n]);
+
+    T_partials_return cdf_ = std_binormal_integral(y1_dbl, y2_dbl, rho_dbl);
+    cdf_log += log(cdf_);
+
+    if (contains_nonconstant_struct<T_y, T_rho>::value) {
+      const T_partials_return inv_cdf_ = 1.0 / cdf_;
+      const T_partials_return one_minus_rho_sq = (1 + rho_dbl) * (1 - rho_dbl);
+      const T_partials_return sqrt_one_minus_rho_sq = sqrt(one_minus_rho_sq);
+      const T_partials_return rho_times_y2 = rho_dbl * y2_dbl;
+      const T_partials_return y1_minus_rho_times_y2 = y1_dbl - rho_times_y2;
+      const bool y1_isfinite = std::isfinite(value_of_rec(y1_dbl));
+      const bool y2_isfinite = std::isfinite(value_of_rec(y2_dbl));
+      const bool y1_y2_arefinite = y1_isfinite && y2_isfinite;
+      const bool rho_lt_one = fabs(rho_dbl) < 1;
+      if (!is_constant_struct<T_y>::value) {
+        ops_partials.edge1_.partials_vec_[n](0)
+            += cdf_ > 0 && rho_lt_one && y1_y2_arefinite
+                   ? inv_cdf_ * exp(std_normal_lpdf(y1_dbl))
+                         * Phi((y2_dbl - rho_dbl * y1_dbl)
+                               / sqrt_one_minus_rho_sq)
+                   : (!y2_isfinite && y1_isfinite && cdf_ > 0)
+                             || (rho_dbl == 1 && y1_dbl < y2_dbl && cdf_ > 0)
+                             || (rho_dbl == -1 && y2_dbl > -y1_dbl && cdf_ > 0)
+                         ? inv_cdf_ * exp(std_normal_lpdf(y1_dbl))
+                         : cdf_ > 0 ? 0 : inv_cdf_;
+        ops_partials.edge1_.partials_vec_[n](1)
+            += cdf_ > 0 && rho_lt_one && y1_y2_arefinite
+                   ? inv_cdf_ * exp(std_normal_lpdf(y2_dbl))
+                         * Phi(y1_minus_rho_times_y2 / sqrt_one_minus_rho_sq)
+                   : (!y1_isfinite && y2_isfinite && cdf_ > 0)
+                             || (rho_dbl == 1 && y2_dbl < y1_dbl && cdf_ > 0)
+                             || (rho_dbl == -1 && y2_dbl > -y1_dbl && cdf_ > 0)
+                         ? inv_cdf_ * exp(std_normal_lpdf(y2_dbl))
+                         : cdf_ > 0 ? 0 : inv_cdf_;
+      }
+      if (!is_constant_struct<T_rho>::value)
+        ops_partials.edge2_.partials_[n]
+            += cdf_ > 0 && y1_y2_arefinite && rho_lt_one
+                   ? inv_cdf_ * 0.5 / (stan::math::pi() * sqrt_one_minus_rho_sq)
```

Comment:
Arithmetic is left associative, so all the constants should be pushed together, here `(0.5 / ...)`.